az-digital/az_quickstart
# Closes #2967, Closes #1138, Closes #3025: Upgrade Drupal core to 10.2.x.

## Description

Upgrades Drupal core to 10.2.x (this will be a 2.9.x minor release only change).

Companion scaffolding repo PR: https://github.com/az-digital/az-quickstart-scaffolding/pull/135
Companion dev metapackage PR: https://github.com/az-digital/az-quickstart-dev/pull/102

### Changes:
- Updates [core SVG logo support patch](https://drupal.org/i/2259567)
- Updates media ~and block~ permissions (see change record: [Media now provides a UI for viewing, reverting and deleting revisions](https://www.drupal.org/node/3366630))
- Adds `showBlocks` plugin to editor configs (see change record: [CKEditor 5 now offers "Show Blocks" functionality](https://www.drupal.org/node/3388811))
- Adds `fetchpriority=high` to site logos (see change record: [Core themes add fetchpriority=high to the site logo, to improve page rendering performance](https://www.drupal.org/node/3373012))
- Applies config differences from re-exporting config using @tadean's [drush command](https://github.com/az-digital/az_quickstart/pull/2540)
  - Related issue: [Give display modes a description](https://www.drupal.org/project/drupal/issues/3364659)


#### 10.2 Compatibility / Deprecation fixes
- Replaces `file_validate_*` functions with `file.validator` plugins (see change record: [file_validate and related functions are deprecated and replaced with file.validator service and Constraint plugins](https://www.drupal.org/node/3363700))
- Adds new constructor argument to forms extending `ConfigFormBase` (see change record: [New parameter added to \Drupal\Core\Form\ConfigFormBase::__construct](https://www.drupal.org/node/3404140))
- Replaces block plugin annotations with block attributes (see change record: [Plugin implementations should use PHP attributes instead of annotations](https://www.drupal.org/node/3395575))

Also includes changes from #3026 to make the card field widget work with upstream field UI changes from Drupal core.

#### Enhancements based on Drupal core changes that will be handled separately
- #3015
- #3016
- #3018
- #3019
- #3011

## Related issues




Closes #2967
Closes #1138 
Closes #3025

## How to test


Link to Probo review site: https://7b5e2985-cedf-4a13-ab91-3a35916c835d--pr-2968.probo.build/

- General testing
- Ensure uploading an SVG logo still works in the theme settings
- Ensure Quickstart settings form and AZ News Feeds admin forms still work
- Ensure author revision pages still work
- Ensure file validation still works for Bibtex import form and footer logo image in theme settings form
- Ensure the Quickstart Carousel Notification block and  Quickstart select menu block still work
- Ensure that `fetchpriority=high` is added to the institutional wordmark and site logo image elements
- Ensure media ~and block~ revisions are accessible for content editors/admins
- Ensure "show blocks" button shows up in ckeditors

### To be tested with separate PRs
- ~Ensure logging still works for config override imports~
- ~Ensure publications import menu page still works~
- ~Ensure Trellis Events import form still works~
- ~Ensure AZ Metrics logging still works~
- ~Ensure AZ News feed importer still works~
- ~Ensure AZ Publications importer still works~
- ~Ensure AZ Core monitoring page route still works~

## Types of changes


### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

### Drupal core
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

### Drupal contrib projects
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

## Checklist
